xonsh/jobs.py

```diff
@@ -170,22 +180,14 @@ def wait_for_active_job(last_task=None, backgrounded=False):
         active_task = get_next_task()
         # Return when there are no foreground active task
         if active_task is None:
-            _give_terminal_to(_shell_pgrp)  # give terminal back to the shell
-            if backgrounded and hasattr(builtins, '__xonsh_shell__'):
-                # restoring sanity could probably be called whenever we return
-                # control to the shell. But it only seems to matter after a
-                # ^Z event. This *has* to be called after we give the terminal
-                # back to the shell.
-                builtins.__xonsh_shell__.shell.restore_tty_sanity()
```

Comment:
see [this comment](https://github.com/xonsh/xonsh/pull/2177#issuecomment-279401696).